app/assets/javascripts/admin/addon/components/admin-flag.gjs

```diff
@@ -0,0 +1,41 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { fn } from "@ember/helper";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import DToggleSwitch from "discourse/components/d-toggle-switch";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default class AdminFlag extends Component {
+  @tracked enabled = this.args.flag.enabled;
+
+  @action
+  toggleFlagEnabled(flag) {
+    this.enabled = !this.enabled;
+
+    return ajax(`/admin/flags/${flag.id}/toggle`, {
+      type: "PUT",
+      contentType: "application/json",
+    }).catch((error) => {
+      this.enabled = !this.enabled;
+      return popupAjaxError(error);
+    });
+  }
+
+  <template>
+    <tr class="flag">
```

Comment:
We should aim for more specificity with CSS classes -- something like `admin-flag-row` in this case would be good. Then you have e.g. `admin-flag-row__name`. This avoids conflicts with other CSS, `flag` is quite a generic class name.